sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -0,0 +1,164 @@
+# Azure.Core public serialization samples
+
+## Using explicit cast
+
+When using protocol methods for advanced handling of RequestContext it is still possible to use the strongly typed models.
+There is an explicit cast operator that can be used to convert the protocol Response to the strongly typed model.
+There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
+
+### Serialization
+
+```C# Snippet:ExplicitCast_Serialize
+PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
+DogListProperty dog = new DogListProperty("myPet");
+Response response = client.CreatePet("myPet", (RequestContent)dog);
+var response2 = client.CreatePet("myPet", RequestContent.Create(dog));
+```
+
+### Deserialization
+
+```C# Snippet:ExplicitCast_Deserialize
+PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
+Response response = client.GetPet("myPet");
+DogListProperty dog = (DogListProperty)response;
+Console.WriteLine(dog.IsHungry);
+```
+
+Given that explicit cast does not allow for serialization options we might also consider a static `FromResponse` and instance `ToRequestContent` methods.
+
+## Using ModelSerializer for NewtonSoftJson
+
+By using the ModelSerializer class, a new instance of Dog does not need to be created before calling Deserialize. Also added ObjectSerializer to Options class so different kinds of Serializers can be used.
+
+### Serialization
+
+```C# Snippet:NewtonSoft_Serialize
+DogListProperty dog = new DogListProperty
+{
+    Name = "Doggo",
+    IsHungry = true,
+    Weight = 1.1,
+    FoodConsumed = { "kibble", "egg", "peanut butter" },
+};
+ModelSerializerOptions options = new ModelSerializerOptions();
+options.Serializers.Add(typeof(DogListProperty), new NewtonsoftJsonObjectSerializer());
+
+BinaryData data = ModelSerializer.Serialize(dog, options);
```

Comment:
Will this be smart enough to set ContentType on BinaryData?

Reply:
Will be addressed in: https://github.com/Azure/azure-sdk-for-net/issues/37708